db/migrate/20191017173513_create_highlights.rb

```diff
@@ -0,0 +1,30 @@
+# OpenStax Highlights
+class CreateHighlights < ActiveRecord::Migration[5.2]
+  def change
+    # https://stackoverflow.com/questions/47064090/rails-postgres-migration-why-am-i-receiving-the-error-pgundefinedfunction
+    enable_extension 'uuid-ossp'
+    enable_extension 'pgcrypto'
+
+    create_table :highlights, id: :uuid do |t|
+      t.uuid :user_uuid, null: false, index: true
+      t.integer :source_type, null: false, index: true, default: 0
+      t.string :source_id, null: false
+      t.jsonb :source_metadata, null: true
+      t.text :source_parent_ids, array: true, default: []
+      t.text :anchor, null: false
+      t.text :highlighted_content, null: false
+      t.text :annotation, null: true
+      t.string :color, null: false
+      t.text :source_order, null: true
+      t.integer :order_in_source, null: true
+      t.jsonb :location_strategies, null: false
+
+      t.timestamps null: false
+    end
+
+    add_index :highlights, :source_parent_ids, using: 'gin'
```

Comment:
Do we need multi-column indexes?  For when we query where user UUID == blah AND source_parent_IDs has blah?

Reply:
yea, we’ll def need those..but i figured we’d wait on that to find out what queries we’d make.  single indexes help a lot (a postgres query would use the first field value’s index and then brute force after that..but that’s usually pretty good!).   optimal is a compound index on exactly the where fields…but i dont know what those are right now…which is why i had just primary fields.  if you know what compound field we are, in the future, going to query on..that’s great if you want to add these now